osu.Game/Storyboards/Commands/IStoryboardCommand.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Transforms;
+using osu.Game.Storyboards.Drawables;
+
+namespace osu.Game.Storyboards.Commands
+{
+    public interface IStoryboardCommand
+    {
+        /// <summary>
+        /// The start time of the storyboard command.
+        /// </summary>
+        double StartTime { get; }
+
+        /// <summary>
+        /// The end time of the storyboard command.
+        /// </summary>
+        double EndTime { get; }
+
+        /// <summary>
+        /// The name of the <see cref="Drawable"/> property affected by this storyboard command.
+        /// Used to apply initial property values based on the list of commands given in <see cref="StoryboardSprite"/>.
+        /// </summary>
+        string PropertyName { get; }
+
+        /// <summary>
+        /// Sets the value of the corresponding property in <see cref="Drawable"/> to the start value of this command.
+        /// </summary>
+        /// <remarks>
+        /// Parameter commands (e.g. <see cref="StoryboardFlipHCommand"/> / <see cref="StoryboardFlipVCommand"/> / <see cref="StoryboardBlendingParametersCommand"/>) only apply the start value if they have zero duration, i.e. take "permanent" effect regardless of time.
```

Comment:
This is gonna require elaboration because I'm not sure what that means. Is this some weird stable quirk? As I understand those commands they flip the drawable / change blending of the drawable between `StartTime` and `EndTime`, so shouldn't a zero-duration "parameter command" do... nothing ever?

Reply:
no, any command with zero duration takes permanent effect

1. the end value is used (not the start value) after the time of the command
2. the start value is used before the time of the command (if there are no commands of the same type before it)